platform/entanglement/src/capabilities/LocationCapability.js

```diff
@@ -0,0 +1,77 @@
+/*global define */
+
+define(
+    function () {
+        "use strict";
+
+        function LocationCapability(domainObject) {
+            this.domainObject = domainObject;
+            return this;
+        }
+
+        /**
+         * Persist the current location of the current domain object as it's
+         * primary location.  Returns a promise.
+         */
+        LocationCapability.prototype.persistLocation = function () {
+            return this.domainObject.useCapability(
+                'mutation',
+                function (model) {
+                    model.location = this.getLocation();
+                }.bind(this)
+            ).then(function () {
+                return this.domainObject
+                    .getCapability('persistence')
+                    .persist();
+            }.bind(this));
+        };
+
+        /**
+         * Return the current location of the current domain object.  Only
+         * valid for domain objects that have a context capability.
+         */
+        LocationCapability.prototype.getLocation = function () {
```

Comment:
The naming of this method is a little ambiguous to me (I'd expect it to return model.location) and I'm not sure if it's needed, aside from its usage for `isLink`

Maybe just have a named function (not a method of LocationCapability) in the script to handle this lookup, e.g.:

```javascript
function contextualLocation(domainObject) {
    var context = domainObject.getCapability('context'),
        parent = context && context.getParent();
    return parent && parent.getId();
}
```

Related note, think it's more useful/accurate to fall back to `undefined` than empty string when there is no context